docs/regression_test_api.rst

```diff
@@ -87,6 +87,40 @@ For instance, continuing with the example above, one could override the :func:`_
        This only has an effect if used with ``inherit_params=True``.
 
 
+.. py:function:: reframe.core.pipeline.RegressionTest.var(name, *types, value=None, field=None)
+
+   Inserts a new regression test variable.
+   If the ``value`` argument is not provided, the variable is considered *declared* but not *defined*.
+   Thus, a variable may not be declared more than once. However, it is possible to alter a variable's value after it was declared by using the :func:`set_var` and :func:`require_var` directives.
+   Note that a variable must be defined before is referenced in the regression test. Otherwise, an :py:exc:`AttributeError` will be raised.
```

Comment:
I see a problem with this, which I think I realised when I was going through the unit tests.  Is the following valid? If not, I think we are making the "required" attribute impractical.

```python
class MyLibTest(rfm.RegressionTest):
    required_var('num_tasks')

class MyTest(MyLibTest):
    def __init__(self):
        self.num_tasks = 10  # and generally take a decision here
```

If I understand correctly, I would have to write my test as follows:

```python
class MyTest(MyLibTest):
    set_var('num_tasks', 1) # new default
    def __init__(self):
        self.num_tasks = 10  # and generally take a decision here
```